cpp/daal/src/algorithms/service_kernel_math.h

```diff
@@ -312,70 +312,54 @@ bool solveEquationsSystemWithPLU(FPType * a, FPType * b, size_t n, size_t nX, bo
     char trans    = 'N';
     DAAL_INT info = 0;
 
-    ScalableMalloc<DAAL_INT, cpu> ipivMalloc;
-    DAAL_INT * ipiv = ipivMalloc.allocate(n);
+    TArrayScalable<DAAL_INT, cpu> ipiv(n);
```

Comment:
Add Check for memory allocation and in other places

Reply:
Fixed